### ISSUE TITLE
Enhancement: hi-res font(s)

### ISSUE THREAD

**Issue by WDLmaster** [jtrfp/terminal-recall#75]
I just remembered that I wanted to re-create the font(s) to be used in TRCL. So I created them from scratch a few minutes ago, still based on the original, but 8 times the resolution. The question is: which format do you prefer for the font? Should I use the NDX+BIN combination (in that case it will be 8 bit, using the game's VGA palette, no alpha-support) or a PNG/TGA/DDS file and yet another text/binary file with font-dimensions? Or a combination of both? It's your choice…

This would replace the tiny font used in briefing text, cockpit display and everything else that is not the bigger, white font. I could re-create that as well but is looks kinda like Arial or something.

**Comment by cjritola** [comment]
Researching/brainstorming; answer in a few days.

**Comment by WDLmaster** [comment]
Sure, no problem. I think I'll also do a hi-res remake of the cockpit texture and other GUI stuff while I'm at it. The stretched, old fashioned 320px texture looks kinda odd on "modern" resolutions...

**Comment by cjritola** [comment]
For the mini text, I can work with a lossless bitmap (png, bmp, pcx) grid of the glyphs so long as there is a horizontal/vertical space between them (no overlap). The higher the res the better.

The big font will be left for v2 since it is technically an enhancement. Here are the results of experimenting with ubiquitous Arial Black.
![big-font-demo](https://cloud.githubusercontent.com/assets/1056698/6000405/7b97f966-aaab-11e4-8914-ddcb12f2e848.png)

**Comment by cjritola** [comment]
ah.
http://www.fontsplace.com/compacta-italic-bt-free-font-download.html
Source:
http://www.flightsimworld.com/forums/topic/119873-flight-simulator-font/

Had a feeling knowing the fact that Randell did FS5 prior to this would pay off.

**Comment by cjritola** [comment]
hmm..
http://en.wikipedia.org/wiki/GPL_font_exception
There is also the question as to whether cleanroom-cloning a font counts as copying or an original-enough creation to be distributed under its own license.

**Comment by WDLmaster** [comment]
On closer inspection, none of the listed and linked fonts look like the one in the game. But I'm almost certain I've seen the "original" font many years ago somewhere. I don't think TRI designed a specific font just for that game.

Here's the in-game charset and it looks actually not even close to Arial. And certainly not to Compacta Italic. But then again, who actually cares if the new font is not 100% accurate. 

![](http://www.wdlmaster.de/TRCL/tvfont.png)

About the recreated font: yes, it is arranged in a grid either way with plenty of space in between (who would do it otherwise?) but you still need the width of every character because it's not monospace AND the order does not follow the ASCII standard. This can be changed, though. Of course you can figure it out by yourself if you wish. The width of every character is always a multiple of 8px, the height is 64px.

If the font is an issue (licensing-wise) what about re-creating other assets? I also did a 2048 * 512 remake of the cockpit textures and wanted to do the other GUI full screen textures as well. But if licensing is an issue, we can't re-create anything because it's all "copied" from the original…

**Comment by cjritola** [comment]
Leaving this feature for consideration in v2.

**Comment by WDLmaster** [comment]
Yes, that's exactly what I did with the mini-font of Fury³ (not vectorized, though). 

To me it seems the bigger white font is assembled from two different fonts. The design between the numbers and the letters is not consistent. The letters are more square but the numbers are completely round. The "8" should almost look like the "B" but is doesn't. And the numbers are smaller. Also, the "6" and the "9" have open lines while the rest of the charset doesn't have that feature – but I should if the design would be consistent…

**Comment by cjritola** [comment]
For version 2 there will be better customization support in place so these can be implemented without depending on me to do it or knowing java.
